modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,724 @@
+            local count = inv.get_item_count(variables.level.demand.item)
+            local price = count * variables.level.demand.price
+            if price ~= 0 then
+                local player = game.players[chest[2]]
+                local balance = Store.get(balances,player)
+                local result = balance + price
+                player.surface.create_entity {
+                    name = "flying-text",
+                    position = ent.position,
+                    text = "+" .. price,
+                    color = {g = 1 }
+                }
+                Store.set(balances,player,result)
+            end
+            inv.clear()
+        end
+    end
+end
+
+local function timer()
+    if started[1] ~= true then
+        local time = variables.level.time
+        time = SecondsToClock((time - (game.tick - variables.tick))/60)
+        if time == "00:00" then
+            started[1] = true
+            variables.tick = game.tick
+            start_game()
+        else
+            for i,player in ipairs(game.connected_players) do
+                local Main_gui = Gui.get_left_element(player, game_gui)
+                local table = Main_gui.container["Money"].table
+                table["Time"].caption = time
+            end
+        end
+    else
+        local time = variables.level.play_time
+        time = SecondsToClock((time - (game.tick - variables.tick))/60)
+        if time == "00:00" then
+            Mini_games.stop_game()
+        end
+        for i,player in ipairs(game.connected_players) do
+            local Main_gui = Gui.get_left_element(player, game_gui)
+            local table = Main_gui.container["Money"].table
+            table["Time"].caption = time
+        end
+    end
+end
+
+token_for_replace_wall =  Token.register(replace_wall)
+Store.watch(balances,function(value,key,_)
+    local player = game.players[key]
+    local Main_gui = Gui.get_left_element(player, game_gui)
+    local table = Main_gui.container["Money"].table
+    table["Balance"].caption = value
+end)
+
+
+local function market(event)
+    local player = game.players[event.player_index]
+    Store.set(balances,player,player.get_item_count("coin"))
+end
+
+local function insideBox(box, pos)
+    local x1 = box[1][1]
+    local y1 = box[1][2]
+    local x2 = box[2][1]
+    local y2 = box[2][2]
+
+    local px = pos.x
+    local py = pos.y
+    return px >= x1 and px <= x2 and py >= y1 and py <= y2
+end
+
+local function player_move(event)
+    local player = game.players[event.player_index]
+    if player.surface.name == variables.level.surface then  --check if the player has not been tped away
+        local center = centers[player.name]
+        if center then
+            local pos = player.position
+            local area = islands[player.name]
+            if not insideBox(area,pos) then
+                player.teleport({center.x,center.y},variables.level.surface)
```

Comment:
Table does not need to be created as a param.
```suggestion
                player.teleport(center, variables.level.surface)
```